### PR TITLE
Forester collecting wood

### PR DESCRIPTION
**Pull request by RichardMXt**
* worker brings wood to hut

### CODE REVIEW

File: Assets/World/Buildings/Lumberjack/Forester/Forester.gd [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+var closest_trees: Array = []
+
+@export var choping_down_tree_time: float = 2
+@export var speed_px_per_sec: float = 64
+
+# Called when the node enters the scene tree for the first time.
+func _ready():
+  set_closest_trees()
+  do_movment_loop()
```
Comment (MichaelXt) [inline Assets/World/Buildings/Lumberjack/Forester/Forester.gd]: movement

File: Assets/World/Buildings/Lumberjack/Forester/Forester.gd [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  var cell_data = tile_map_layer.get_cell_tile_data(tile_map_layer.local_to_map(tree_pos))
+  var is_there_still_a_tree: bool = cell_data != null and cell_data.get_custom_data(tile_map_layer.is_tree)
+
+  return is_there_still_a_tree and not tile_map_layer.trees_getting_choped.has(tree_pos)
+
+
+func do_movment_loop():
+  await wait_for_tree_in_need()
+
+  var tree_pos = await go_to_tree()
```
Comment (MichaelXt) [inline Assets/World/Buildings/Lumberjack/Forester/Forester.gd]: while not cancellation_requested:
await storage_for_the_tree_available()
tree_pos = get_closest_available_tree()
if tree_pos == null: sleep(1s); continue
mark_tree_for_chopping(tree_pos)
await walk_to_tree()
await chopdown_tree()
await walk_back_and_unload()

File: Assets/World/Buildings/Lumberjack/Forester/ForesterMovmentFrames.tres [hunk 3]
```diff
@@ -0,0 +1,341 @@
+[gd_resource type="SpriteFrames" load_steps=65 format=3 uid="uid://b7inbywvvhts2"]
+
+[ext_resource type="Texture2D" uid="uid://7hnp6hwaxcsv" path="res://Assets/World/Buildings/Lumberjack/Sprites/Move/0/0001.png" id="1_xxwri"]
```
Comment (MichaelXt) [inline Assets/World/Buildings/Lumberjack/Forester/ForesterMovmentFrames.tres]: combine all animation frames to a single sprite

**Review by MichaelXt — CHANGES_REQUESTED** [review]

File: Assets/World/Buildings/Lumberjack/Forester/ForesterMovmentFrames.tres (hunk 3 above)
Comment (RichardMXt) [reply Assets/World/Buildings/Lumberjack/Forester/ForesterMovmentFrames.tres]: done

File: Assets/World/Buildings/Lumberjack/Forester/Forester.gd (hunk 2 above)
Comment (RichardMXt) [reply Assets/World/Buildings/Lumberjack/Forester/Forester.gd]: done

File: Assets/World/Buildings/Lumberjack/Forester/Forester.gd (hunk 1 above)
Comment (RichardMXt) [reply Assets/World/Buildings/Lumberjack/Forester/Forester.gd]: done

**Pull request merged**